### DISCUSSION TITLE
Support for altitude data in 3D view

### DISCUSSION THREAD

**Discussion by demiantres** [mapsforge/vtm#1160] [Q&A] [answered]
Some mapsforge maps (like OpenAndroMaps) have altitude data included. In the 3D view it would be nice to have an option to visualize the altitude. As an alternative source also HGT files could be used.

**Comment by devemux86** [comment]
VTM is able to blend regular tiles sources (e.g. Mapsforge maps) with hillshading (raster) tile sources (e.g. Hike & Bike). See `BitmapTileTest` example in `vtm-playground` (press `3`).

What you're asking is additionally to visualize the terrain as actual 3D in Z axis?

**Comment by demiantres** [comment]
Exactly, a terrain visualization option.

**Comment by sunshine0576** [comment]
Height map like this
![image](https://user-images.githubusercontent.com/21358135/45675700-7fcb0200-bb62-11e8-9140-e99701f200c2.png)

**Comment by devemux86** [comment]
Can already blend regular maps and [relief maps](https://wiki.openstreetmap.org/wiki/Relief_maps) in VTM:

![vtm](https://user-images.githubusercontent.com/3484020/45676724-171c5f00-bb3b-11e8-88f0-0923f8eeb18f.png)

**Comment by sunshine0576** [comment]
Does the data need to be rendered and processed by itself?（For example, I configure grid data through QGIS, and then publish it to XYZ Tiles through Geoserver.） 

Can it be used directly in VTM ? I have not discovered that the existing layer supports this function.

**Comment by devemux86** [comment, marked as answer]
VTM does not support currently Hillshading from digital elevation model data like [HGT](http://wiki.openstreetmap.org/wiki/SRTM#Format) format.
That works in Mapsforge with a different engine, processing bitmap images (https://github.com/mapsforge/mapsforge/issues/923).

I mentioned [above](https://github.com/mapsforge/vtm/issues/197#issuecomment-251876010) the example of how to blend **pre-rendered** shaded relief maps in VTM.
